### PR TITLE
obsolete unit of work

### PR DESCRIPTION
**Pull request by adamralph**
- [x] revert changes to mutators
- [x] WARN in 8, ERROR in 9, and remove in 10

### CODE REVIEW

File: src/NServiceBus.Core/obsoletes-v8.cs [hunk 1]
```diff
@@ -806,4 +806,44 @@ public interface ICancelDeferredMessages
     }
 }
 
+namespace NServiceBus.MessageMutator
+{
+    using System.Threading.Tasks;
+
+    [ObsoleteEx(
+        Message = "Use behaviors instead",
+        TreatAsErrorFromVersion = "8.0.0",
```
Comment (andreasohlund) [inline src/NServiceBus.Core/obsoletes-v8.cs:815]: Should be `TreatAsErrorFromVersion = "9.0.0",` to give a WARN in v8 right?

Comment (adamralph) [reply src/NServiceBus.Core/obsoletes-v8.cs:815]: 👍 we'll follow whatever we agree on in https://github.com/Particular/Platform/pull/10

Comment (adamralph) [reply src/NServiceBus.Core/obsoletes-v8.cs:815]: 👍 we agreed to WARN in NServiceBus 8, ERROR in NServiceBus 9, and remove in NServiceBus 10.

File: src/NServiceBus.Core/UnitOfWork/IManageUnitsOfWork.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@ namespace NServiceBus.UnitOfWork
     /// Interface used by NServiceBus to manage units of work as a part of the
     /// message processing pipeline.
     /// </summary>
+    [ObsoleteEx(ReplacementTypeOrMember = "Behavior", TreatAsErrorFromVersion = "9", RemoveInVersion = "10")]
```
Comment (adamralph) [inline src/NServiceBus.Core/UnitOfWork/IManageUnitsOfWork.cs:10]: @andreasohlund @DavidBoike I wasn't really sure what to write for `ReplacementTypeOrMember` and whether we should also provide a `Message`.

Same applies to the other two types.

Comment (DavidBoike) [reply src/NServiceBus.Core/UnitOfWork/IManageUnitsOfWork.cs:10]: Just gets concatenated into the generated Obsolete attribute. What about this?
```suggestion
    [ObsoleteEx(Message = "The unit of work pattern is more straightforward to implement in a pipeline behavior, where the using keyword and try/catch blocks can be used.", ReplacementTypeOrMember = "Behavior<TContext>", TreatAsErrorFromVersion = "9", RemoveInVersion = "10")]
```

**Review by DavidBoike — APPROVED** [review]

**Pull request merged**